api/data/info.go

```diff
@@ -68,6 +68,15 @@ func (b *BucketInfo) CORSObjectName() string { return bktCORSConfigurationObject
 // Version returns object version from ObjectInfo.
 func (o *ObjectInfo) Version() string { return o.ID.String() }
 
+// NullableVersion returns object version from ObjectInfo.
+// Return "null" if "S3-Versions-unversioned" header present.
+func (o *ObjectInfo) NullableVersion() string {
```

Comment:
I think, the `Nullable` prefix seems redundant a bit. 

As I see `Version` function is used only in `NullableVersion` and `TagsObject`. Maybe it'd be more clear to write something like this: 
```
// Version returns object version from ObjectInfo.
// Return "null" if "S3-Versions-unversioned" header present.
func (o *ObjectInfo) Version() string {
	if _, ok := o.Headers["S3-Versions-unversioned"]; ok {
		return "null"
	}
	return o.o.ID.String()
}
...
func (o *ObjectInfo) TagsObject() string { return ".tagset." + o.Name + "." + o.ID.String() }
```

Reply:
In most cases we want to treat version as object id (for storing in headers and so on). `Version()` function is used not only in `NullableVesion` and `TagsObject` (we use it outside `data` package). But currently `NullableVersion` is used only once (to encode `list-object-versions` response) so we can define similar function there and drop it from this package

Reply:
I think it is okay to keep it there, though. You may want to specify in the comment, that `NullableVersion` is required for listing object version.